samples/bluetooth/opendroneid/src/main.c

```diff
@@ -0,0 +1,281 @@
+}
+
+static void update_payload(uint8_t turn)
+{
+	int err = 0;
+
+	switch (turn) {
+	case 0: /* BasicID */
+		err = encodeBasicIDMessage((ODID_BasicID_encoded *)&encoded, &uasData.BasicID[0]);
+		if (err == ODID_SUCCESS) {
+			memcpy(&payload[4], &encoded, sizeof(ODID_BasicID_encoded));
+			memcpy(&payload[3], &msg_counters[ODID_MSG_COUNTER_BASIC_ID], 1);
+			++msg_counters[ODID_MSG_COUNTER_BASIC_ID];
+		}
+		break;
+	case 1: /* Location */
+		/* Updating location for checking whether messages are dropped or not. */
+		uasData.Location.Latitude = uasData.Location.Latitude - 0.01;
+		uasData.Location.Longitude = uasData.Location.Longitude + 0.01;
+		err = encodeLocationMessage((ODID_Location_encoded *)&encoded, &uasData.Location);
+		if (err == ODID_SUCCESS) {
+			memcpy(&payload[4], &encoded, sizeof(ODID_Location_encoded));
+			memcpy(&payload[3], &msg_counters[ODID_MSG_COUNTER_LOCATION], 1);
+			++msg_counters[ODID_MSG_COUNTER_LOCATION];
+		}
+		break;
+	case 2: /* Auth */
+		err = encodeAuthMessage((ODID_Auth_encoded *)&encoded, &uasData.Auth[0]);
+		if (err == ODID_SUCCESS) {
+			memcpy(&payload[4], &encoded, sizeof(ODID_Auth_encoded));
+			memcpy(&payload[3], &msg_counters[ODID_MSG_COUNTER_AUTH], 1);
+			++msg_counters[ODID_MSG_COUNTER_AUTH];
+		}
+		break;
+	case 3: /* SelfID */
+		err = encodeSelfIDMessage((ODID_SelfID_encoded *)&encoded, &uasData.SelfID);
+		if (err == ODID_SUCCESS) {
+			memcpy(&payload[4], &encoded, sizeof(ODID_SelfID_encoded));
+			memcpy(&payload[3], &msg_counters[ODID_MSG_COUNTER_SELF_ID], 1);
+			++msg_counters[ODID_MSG_COUNTER_SELF_ID];
+		}
+		break;
+	case 4: /* System */
+		err = encodeSystemMessage((ODID_System_encoded *)&encoded, &uasData.System);
+		if (err == ODID_SUCCESS) {
+			memcpy(&payload[4], &encoded, sizeof(ODID_System_encoded));
+			memcpy(&payload[3], &msg_counters[ODID_MSG_COUNTER_SYSTEM], 1);
+			++msg_counters[ODID_MSG_COUNTER_SYSTEM];
+		}
+		break;
+	case 5: /* OperatorID */
+		err = encodeOperatorIDMessage((ODID_OperatorID_encoded *)&encoded,
+					      &uasData.OperatorID);
+		if (err == ODID_SUCCESS) {
+			memcpy(&payload[4], &encoded, sizeof(ODID_OperatorID_encoded));
+			memcpy(&payload[3], &msg_counters[ODID_MSG_COUNTER_OPERATOR_ID], 1);
+			++msg_counters[ODID_MSG_COUNTER_OPERATOR_ID];
+		}
+		break;
+	default:
+		break;
+	}
+}
+
+int main(void)
+{
+	int err;
+
+	printf("Starting ODID Demo\n");
+
+	/* Initialize UAS data. */
+	odid_initUasData(&uasData);
+	fill_example_data(&uasData);
+	fill_example_gps_data(&uasData);
+	memset(&encoded, 0, sizeof(union ODID_Message_encoded));
+	for (int i = 0; i < ODID_MSG_COUNTER_AMOUNT; ++i) {
+		msg_counters[i] = 0;
+	}
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(bt_ready);
+	if (err) {
+		printf("Bluetooth init failed (err %d)\n", err);
+		return -1;
+	}
+
+	while (true) {
+		if (bt_is_ready()) {
+			break;
+		}
+
+		printf("Bluetooth not ready. Checking again in 100 ms\n");
+		k_sleep(K_MSEC(100));
+	}
```

Comment:
This does't look particularly elegant. You could instead have this wait on a semaphore that your `bt_ready()` callback eventually gives, however the `bt_enable()` API already has support for blocking behavior: you just need to pass NULL to it, and then do the stuff you do in your `bt_ready()` right here in `main()`